### PR TITLE
Receiving sms dev

### PR DESCRIPTION
**Pull request by Blazej-Zielinski**

### CODE REVIEW

**Review by filip6464 — APPROVED** [review]
legit, looks okay

**Review by wiktorwieclaw — APPROVED** [review]
This feature was implemented together live using Discord and CLion's CodeWithMe and all the code was verified in real time.

File: UE/Application/Ports/IBtsPort.hpp [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #pragma once
 
 #include "Messages/BtsId.hpp"
+#include "Sms.hpp"
```
Comment (rafauu) [inline UE/Application/Ports/IBtsPort.hpp]: use fwd

File: UE/Application/Sms.hpp [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include "Messages/PhoneNumber.hpp"
```
Comment (rafauu) [inline UE/Application/Sms.hpp]: include string

File: UE/Application/States/ConnectedState.hpp [hunk 3]
```diff
@@ -13,6 +13,7 @@ class ConnectedState : public BaseState
     // IBtsEventsHandler interface
 public:
     void handleDisconnected() final;
+    void handleSms(const Sms& sms) final;
```
Comment (rafauu) [inline UE/Application/States/ConnectedState.hpp:16]: fwd

File: UE/Application/States/BaseState.hpp [hunk 4]
```diff
@@ -3,6 +3,7 @@
 #include "IEventsHandler.hpp"
 #include "Logger/PrefixedLogger.hpp"
 #include "Context.hpp"
+#include "Sms.hpp"
```
Comment (rafauu) [inline UE/Application/States/BaseState.hpp]: fwd

File: UE/Application/CMakeLists.txt [hunk 5]
```diff
@@ -5,5 +5,5 @@ aux_source_directory(. SRC_LIST)
 aux_source_directory(Ports SRC_LIST)
 aux_source_directory(States SRC_LIST)
 
-add_library(${PROJECT_NAME} ${SRC_LIST})
+add_library(${PROJECT_NAME} ${SRC_LIST} SmsDb.cpp ISmsDb.hpp)
```
Comment (rafauu) [inline UE/Application/CMakeLists.txt:8]: revert

File: UE/Application/ISmsDb.hpp [hunk 6]
```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include "Messages/PhoneNumber.hpp"
+
+namespace ue {
+
+struct Sms
```
Comment (rafauu) [inline UE/Application/ISmsDb.hpp:7]: extract to separate file and leave fwd Sms

File: UE/Application/ISmsDb.hpp [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+{
+    common::PhoneNumber from;
+    std::string text;
+
+    Sms(common::PhoneNumber from, std::string text)
+            : from{from}, text{std::move(text)} {}
+};
+
+inline bool operator==(const Sms& lhs, const Sms& rhs) {
+    return lhs.from == rhs.from && lhs.text == rhs.text;
```
Comment (rafauu) [inline UE/Application/ISmsDb.hpp:17]: info
member fun - auto tie() { return tuple{from, text}; }
operator== { lhs.tie() == rhs.tie(); }

cpp20 - spaceship operator <=>

File: UE/Application/Ports/IBtsPort.hpp [hunk 8]
```diff
@@ -1,7 +1,7 @@
 #pragma once
 
 #include "Messages/BtsId.hpp"
-#include "Sms.hpp"
+#include "SmsDb.hpp"
```
Comment (rafauu) [inline UE/Application/Ports/IBtsPort.hpp:4]: fwd

File: UE/Application/States/BaseState.hpp [hunk 9]
```diff
@@ -3,7 +3,7 @@
 #include "IEventsHandler.hpp"
 #include "Logger/PrefixedLogger.hpp"
 #include "Context.hpp"
-#include "Sms.hpp"
+#include "SmsDb.hpp"
```
Comment (rafauu) [inline UE/Application/States/BaseState.hpp:6]: fwd

File: UE/Tests/Application/CMakeLists.txt [hunk 10]
```diff
@@ -7,7 +7,7 @@ aux_source_directory(Ports SRC_LIST)
 include_directories(${COMMON_DIR}/Tests)
 include_directories(${UE_DIR}/Tests)
 
-add_executable(${PROJECT_NAME} ${SRC_LIST})
+add_executable(${PROJECT_NAME} ${SRC_LIST} Mocks/ISmsDbMock.hpp Mocks/ISmsDbMock.cpp SmsDbTestSuite.cpp)
```
Comment (rafauu) [inline UE/Tests/Application/CMakeLists.txt:10]: remove

File: UE/Tests/Application/Mocks/ISmsDbMock.hpp [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+
+#include "ISmsDb.hpp"
+
+namespace ue
+{
+
+class ISmsDbMock : public ISmsDb
+{
+public:
+    ISmsDbMock() = default;
```
Comment (rafauu) [inline UE/Tests/Application/Mocks/ISmsDbMock.hpp:13]: you can move definition to cpp

File: UE/Tests/Application/SmsDbTestSuite.cpp [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+#include "SmsDb.hpp"
+#include "Messages/PhoneNumber.hpp"
+
+namespace ue {
+
+using namespace ::testing;
+
+class SmsDbTestSuite : public Test {
+protected:
+    const common::PhoneNumber PHONE_NUMBER{112};
```
Comment (rafauu) [inline UE/Tests/Application/SmsDbTestSuite.cpp:12]: static constexpr
or
move to anon ns

**Pull request merged**